packages/trace-viewer/src/ui/aiConversation.tsx

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useState } from 'react';
+import Markdown from 'react-markdown'
+import './aiConversation.css';
+import { clsx } from '@web/uiUtils';
+import { Conversation, LLMMessage } from '@isomorphic/llm';
+
+export function AIConversation({ history, conversation, firstPrompt }: { history: LLMMessage[], conversation: Conversation, firstPrompt?: LLMMessage }) {
+  const [input, setInput] = useState('');
+  const [abort, setAbort] = useState<AbortController>();
+
+  const send = useCallback(async (prompt: string, visiblePrompt?: string) => {
+    const controller = new AbortController();
+    setAbort(controller);
+    try {
+      await conversation.send(prompt, visiblePrompt, controller.signal);
+    } finally {
+      setAbort(undefined);
+    }
+  }, [conversation]);
+
+  const onSubmit = useCallback(async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setInput('');
+    const content = new FormData(event.target as any).get('content') as string;
+    await send(content);
+  }, [send]);
+
+  useEffect(() => {
+    if (!conversation.isEmpty())
+      return;
+    if (!firstPrompt)
+      return;
+
+    send(firstPrompt.content, firstPrompt.displayContent);
```

Comment:
Does this automatically send something over the API without the user clicking anywhere? Looks suspicious :)